include/bluetooth/conn.h

```diff
@@ -227,13 +227,11 @@ int bt_le_set_auto_conn(bt_addr_le_t *addr,
  *
  *  Returns a new reference that the the caller is responsible for managing.
  *
- *  @param peer  Remote address.
  *  @param param Directed advertising parameters.
  *
  *  @return Valid connection object on success or NULL otherwise.
  */
-struct bt_conn *bt_conn_create_slave_le(const bt_addr_le_t *peer,
-					const struct bt_le_adv_param *param);
+struct bt_conn *bt_conn_create_slave_le(const struct bt_le_adv_param *param);
```

Comment:
Why are you moving the peer_addr into the options? Is it because you're calling `bt_le_adv_start()` internally and that API does not take a peer_addr parameter? In that case I'd refactor bt_le_adv_start() to a public and an internal function. The public stays as-is and the internal one takes a peer_addr (which can be NULL). This also eliminates the issue with breaking the BT_LE_ADV_OPT() macro for existing users.

Reply:
That's one reason. Apart from that, peer address is considered to be an Advertising Parameter according to the Bluetooth specification (like ```interval_min```, ```interval_max```), so I thought it would be okay to put it in the ```bt_le_adv_param``` structure.

I don't have any strong opinion on this, so I will go along with your suggestion

Reply:
I understand your reasoning. However, it'd be nice not to impose the 4 bytes overhead on most users of the API who don't use directed advertising. Also, having the address in the adv_param would raise the question how `bt_le_adv_start()` should behave with a non-NULL peer_addr if called directly by the app instead of through bt_conn_create_slave_le(). I think it's something we'd want to prevent, i.e. it should probably have returned an error. Even with the new design I think `bt_le_adv_start()` should return an error if called with the new advertising option set.

Reply:
If you refactor the code to an internal `static int adv_start(...)` function that both `bt_conn_create_slave_le()` and `bt_le_adv_start()` call, then then `bt_le_adv_start()` could check for the flag before calling the internal function. Telling the app it did something wrong instead of silently ignoring an explicitly given option seems like more helpful policy toward the developer. Another alternative could be an assert.

Btw, with the single-bit approach, I'd make the default high-duty and call the new option LOW_DUTY, simply because high-duty came first in the core spec versions.